### PR TITLE
Update test strategy

### PR DESCRIPTION
**Pull request by 4ydan**
## Basics

- [ ] I added a line to [/doc/CHANGELOG.md](/doc/CHANGELOG.md)
- [ ] The PR is rebased with current master.
- [ ] Details of what you changed are in commit messages.
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildserver is happy.

## Checklist

- [ ] I have installed and I am using [pre-commit hooks](../doc/contrib/README.md#Hooks)
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I fixed the introduction tour
- [ ] I wrote migrations in a way that they are compatible with already present data
- [ ] I fixed all affected decisions
- [ ] I added unit tests for my code
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](https://github.com/ElektraInitiative/PermaplanT/blob/master/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [ ] I created left-over issues for things that are still to be done
- [ ] Code is conforming to [our Architecture](/doc/architecture)
- [ ] Code is conforming to [our Guidelines](/doc/guidelines)
      (exceptions are documented)
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] Examples are well chosen and understandable

### CODE REVIEW

**Comment by 4ydan** [comment]
Thank you for your valuable feedback. I do not know much about backend/frontend testing for now, so I sadly can not answer your async question. Maybe @Bushuo knows more about this.

File: doc/tests/README.md [hunk 1]
```diff
@@ -66,6 +66,30 @@ mod tests {
...
 ```
 
+#### Frontend
+
+For the frontend unit tests will be used to test the following areas:
+
+- State management logic: Since the frontend relies heavily on state management, it is important to ensure that the state is being properly managed and updated.
+- Functions that handle data manipulation, such as sorting or filtering
+- Components that render UI elements
+- Async actions, such as API calls or event handling
```
[the remark this answers is not in the document]
Comment (markus2330) [reply doc/tests/README.md:76]: Can API calls even be tested using unit tests in the frontend?

```suggestion
- Correct event handling
```

File: doc/tests/README.md [hunk 2]
```diff
@@ -92,8 +113,9 @@ The whole module is annotated with `#[cfg(test)]` and will therefore only be com
 System testing will include end-to-end testing of the application to ensure overall functionality and user experience.
 This will include testing of all features, navigation, and error handling.
 
-- Test scenarios must be derived from functional requirements and use cases.
-- Testers must perform End-to-End testing of the software, including all user workflows and system integrations.
+- System test scenarios must be derived from functional requirements and use cases.
```
Comment (markus2330) [inline doc/tests/README.md:116]:
```suggestion
- System test scenarios must be derived from the use cases and the requirements they imply.
```

File: doc/tests/README.md [hunk 3]
```diff
@@ -92,8 +113,9 @@ The whole module is annotated with `#[cfg(test)]` and will therefore only be com
...
+- Manual system tests are documented in [`doc/tests/manual/protocol.md`](./manual/protocol.md).
+- Automated system tests (e2e tests) can be found in [`e2e/`](../../e2e/README.md).
```
Comment (markus2330) [inline doc/tests/README.md:118]: What about something like? Please add link if possible.

```suggestion
- Automated system tests (e2e tests) can be found in the top-level `e2e` folder.
- Both are specified in Gherkin syntax.
```

**Review by markus2330 — APPROVED** [review]
Great job! :herb: A link is broken (see CI), otherwise it looks fine for me.

File: doc/tests/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (markus2330) [reply doc/tests/README.md:76]: Ahh, ok, I think we have slightly different terminology here. For me unit tests in frontend was component/function testing of a single component/function. Testing how several components/functions work together in the frontend is already integration testing (afaik not present yet). But also integration testing is executed by a unit testing framework.

Using the frontend+backend together is only done in e2e tests.

See also testing section in doc/architecture/12glossary.md (which actually was written for backend and should be updated to also include an example for the frontend).

@tschawax I think the first step is that you look at which frontend tests we currently have and if they already do more or less what is described in the strategy here and how we go from where we are. Of course you can also recommend improvements of the strategy.

**Pull request merged**